src/sidebar/util/test/observe-element-size-test.js

```diff
@@ -2,9 +2,28 @@
 
 const observeElementSize = require('../observe-element-size');
 
+/**
+ * Wait for a condition to become true.
+ *
+ * @param {() => boolean} callback
+ */
+function waitFor(callback) {
+  return new Promise(resolve => {
+    const timer = setInterval(() => {
+      if (callback()) {
+        clearInterval(timer);
+        resolve();
+      }
+    }, 0);
+  });
+}
+
 /**
  * Give MutationObserver, ResizeObserver etc. a chance to deliver their
  * notifications.
+ *
+ * This waits for a fixed amount of time. If you can wait for a specific event
```

Comment:
I can't quite make sense of this comment for some reason :).

Reply:
What I meant was that if you want to check that some event happens "eventually", it is better for the code to wait until that specific event is detected, rather than wait a fixed amount of time and then check if the event happened. Does that make sense?